doc/sai-ptf/lag_test_plan.md

```diff
@@ -0,0 +1,140 @@
+# SAI Lag Test plan
+- [SAI Lag Test plan](#sai-lag-test-plan)
+  - [Overriew](#overriew)
+    - [Test Topology](#test-topology)
+    - [Testbed](#testbed)
+  - [Scope](#scope)
+- [Basic SAI APIs and sample packets](#basic-sai-apis-and-sample-packets)
+  - [APIs](#apis)
+  - [Packets](#packets)
+- [Test suites](#test-suites)
+  - [Test suite #1: PortChannel Loadbalanceing](#test-suite-1-portchannel-loadbalanceing)
+  - [Test suite #2: Ingress/Egreee disable](#test-suite-2-ingressegreee-disable)
+## Overriew
+The purpose of this test plan is to test the LAG/PortChannel function from SAI.
+
+### Test Topology
+For SAI-PTF, it will use a non-topology network structure for the sai testing. 
+
+### Testbed
+Those tests will be run on the testbed structure, the components are:
+* PTF - running in a server that can connect to the target DUT
+* SAI server - running on a dut
+
+*p.s. cause the SAI testing will not depend on any sonic components, then there will be no specific topology(T0 T1 T2) for testing.*
+
+## Scope
+The test will include two parts
+1. Lag functionalities
+   - Load balancing
+2. Lag SAI APIs
+   - create/check/remove lag and lag member
+
+
+# Basic SAI APIs and sample packets
+
+## APIs
+
+Create and lag member
+```Python
+sai_thrift_create_lag(self.client)
+sai_thrift_create_lag_member(
+            self.client, lag_id=lag3, port_id=self.port24)
+```
+
+Get lag members
+```Python
+sai_thrift_get_lag_attribute(
+            self.client, lag3, port_list=portlist)
+count = attr_list["SAI_LAG_ATTR_PORT_LIST"].count
+```
+
+Get port counter
+```Python
+counter_results = sai_thrift_get_port_stats(client, port)
+
+counter_results["SAI_PORT_STAT_IF_IN_DISCARDS"],
+counter_results["SAI_PORT_STAT_IF_IN_DISCARDS"],
+counter_results["SAI_PORT_STAT_IF_IN_UCAST_PKTS"],
+counter_results["SAI_PORT_STAT_IF_OUT_UCAST_PKTS"]))
+```
+
+Add fdb entry
+**P.s.For fordwarding packet with vlan tag, fdb entry should enable the bv_id**
+```python
+sai_thrift_fdb_entry_t(switch_id=self.switch_id, mac_address=mac1, bv_id=self.vlan_oid)
+sai_thrift_create_fdb_entry(
+            self.client,
+            fdb_entry,
+            type=SAI_FDB_ENTRY_TYPE_STATIC,
+            bridge_port_id=port_bp,
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+```
+
+Remove lag member
+```Python
+sai_thrift_remove_lag_member(self.client,  self.lag1_member6)
+```
+
+## Packets
+Vlan tagged packet
+```Python
+simple_udp_packet(eth_dst=dst_mac,
+                    eth_src=src_mac,
+                    dl_vlan_enable=True,
+                    vlan_vid=vlan_id,
+                    pktlen=104)
+```
+
+# Test suites
+## Test suite #1: PortChannel Loadbalanceing
```

Comment:
For this one, maybe we can divide them to 2 testcases, one is focus on the load balance. The other is for add/delete operation, which has no pkt loss.

For the load balance, maybe we also need to test the algorithm selection for the hash.